docroot/modules/custom/mass_entity_usage/tests/src/ExistingSite/EntityUsageTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace Drupal\Tests\mass_entity_usage\ExistingSite;
+
+use Drupal\file\Entity\File;
+use Drupal\mass_content_moderation\MassModeration;
+use Drupal\node\Entity\Node;
+use Drupal\user\Entity\User;
+use DrupalTest\QueueRunnerTrait\QueueRunnerTrait;
+use Drush\Drupal\Commands\core\QueueCommands;
+use Drush\Log\DrushLoggerManager;
+use weitzman\DrupalTestTraits\Entity\MediaCreationTrait;
+use weitzman\DrupalTestTraits\ExistingSiteBase;
+use weitzman\LoginTrait\LoginTrait;
+
+/**
+ * Class EntityUsageTest.
+ */
+class EntityUsageTest extends ExistingSiteBase {
+
+  use LoginTrait;
+  use MediaCreationTrait;
+  use QueueRunnerTrait;
+
+  private $user;
+
+  /**
+   * Create the user.
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->emptyEntityUsageQueues();
+    $user = User::create(['name' => $this->randomMachineName()]);
+    $user->addRole('administrator');
+    $user->activate();
+    $user->save();
+    $this->user = $user;
+    $this->drupalLogin($user);
+  }
+
+  /**
+   * Counts how many usage rows an entity has on its /mass-usage page.
+   */
+  private function countUsageRows($entity) {
+    /** @var \Drupal\Core\Entity\EntityBase $entity */
+    $this->drupalGet($entity->getEntityType()->id() . '/' . $entity->id() . '/mass-usage');
+    $table = $this->getCurrentPage()->find('css', 'main table');
+    if ($table->find('css', 'td.empty.message')) {
+      return 0;
+    }
+    return count($table->findAll('css', 'tbody tr td a'));
+  }
+
+  /**
+   * Asserts how many usage rows an entity has on its /mass-usage page.
+   */
+  private function assertUsageRows($entity, $usage_expected) {
+    $usage_actual = $this->countUsageRows($entity);
+    $this->assertEquals($usage_expected, $usage_actual, 'Usage is ' . $usage_actual . ' instead of ' . $usage_expected);
+  }
+
+  /**
+   * Creates 1 media, 1 org_page and 1 curated_list referencing the org_page.
+   */
+  private function createEntities($curated_list_state) {
+    // Create a file to upload.
+    $destination = 'public://file-001.txt';
+    $file = File::create([
+      'uri' => $destination,
+    ]);
+    $file->setPermanent();
+    $file->save();
+    $src = 'modules/custom/mass_entity_usage/tests/fixtures/file-001.txt';
+
+    /** @var \Drupal\Core\File\FileSystemInterface $file_system */
+    $file_system = \Drupal::service('file_system');
+    $file_system->copy($src, $destination, TRUE);
+
+    // Create media item.
+    $media = $this->createMedia([
+      'title' => 'Llama',
+      'bundle' => 'document',
+      'field_upload_file' => [
+        'target_id' => $file->id(),
+      ],
+      'status' => 1,
+    ]);
+
+    $node_org = $this->createNode([
+      'type' => 'org_page',
+      'title' => 'Test Organization',
+      'uid' => $this->user->id(),
+      'moderation_state' => MassModeration::PUBLISHED,
+    ]);
+
+    // Entity usage tracking is done at the end of the requests, so we need to
```

Comment:
This is no longer true. We process in a queue now. Not sure how that affects the code.

Reply:
It doesn't affect the test. I will remove the comment.